_episodes/01-package-setup.md

```diff
@@ -235,44 +240,51 @@ from .functions import *
+goes to the `functions.py` file, and brings everything that is defined there into the file. When we use our function defined in `functions.py`, that means we will be able to just say `molecool.canvas()` instead of giving the full path `molecool.functions.canvas()`. If that's confusing, don't worry too much for now. We will be returning to `__init__.py` in a few minutes. For now, just note that it exists and makes our directory into a package.
 
 ### Our first module
-Once inside of the `molecool` folder (`molecool/molecool`), examine the files that are there. View the first module (`functions.py`) in a text editor. We see a few things about this file. The top begins with a description of this module surrounded by three quotations (`"""`). Right now, that is the file name, followed by our short description, then the sentence "Handles the primary functions". We will change this to be more descriptive later. CookieCutter has also created a placeholder function in called `canvas`.  At the start of the `canvas` function, we have a `docstring` (more about this in [documentation]), which describes the function.
+Once inside the `molecool` folder (`molecool/molecool`), examine the files that are there. View the module (`functions.py`) in a text editor. We see a few things about this file. The top begins with a description of this module surrounded by three quotations (`"""`). Right now, that is the file name, followed by our short description, then the sentence "Handles the primary functions". We will change this to be more descriptive later. CookieCutter has also created a placeholder function called `canvas`.  At the start of the `canvas` function, we have a `docstring` (more about this in [documentation]), which describes the function.
+
+We will be moving all of the functions we defined in the Jupyter notebook into python modules (`.py` files) like these.
 
-We will be moving all of the functions we defined in the jupyter notebook into python modules (`.py` files) like these.
+### Installing from local source.
 
-### Python local installs
+You may be accustomed to `pip` automatically retrieving packages from the internet. You can also install packages from local sources that contain a `setup.py` file.
 
-To develop this package, we will want to something called a developmental install so that we can try out our functions and package as we develop it. 
+To develop this package, we will want to use what is called "development mode" or an "editable install" so that we can try out our functions and package as we develop it. We access development mode using the `develop` command to `setup.py`, or the `-e` option to `pip`.
+
+*TODO: Note that "editable" install is not (yet) standard and may even go away in the future.*
```

Comment:
Can you add a reference for this?

Reply:
From what I know calling setup.py directly (for [install](https://mobile.twitter.com/pganssle/status/1152694289920679936), [test](https://github.com/pytest-dev/pytest/issues/5534), etc.) is deprecated. Pip comes bundled with Python 3.4+ so let's stick with pip. And as a side note from what I learned, Poetry install your local package in editable mode as default, CMIIW.

@eirrgang could you elaborate more what does it mean by "not (yet) a standard"? From what I know there are lots of way to develop and test your project depending on what kind of project that you working on. For example in Web app framework like Django you can use the demo server. Or you can also directly deploy your code to a docker container every time you make the change. But I think installing in editable mode gives the freedom to check the API, or function/module without having to uninstall and install the package that we develop every time we make the change, which is very practical.

Reply:
There have been rumblings about deprecating non PEP517/518 behaviors, and I was thinking of the note at https://setuptools.readthedocs.io/en/latest/setuptools.html#setup-cfg-only-projects

But it is probably just something to keep an eye on, and doesn't warrant an update to the material at this time.

`setuptools` (and `pip`) are presenting an increasingly unified and standardized approach with the PEPs (wheel vs. egg, etc). We should definitely stick with `pip`. But you can't read documentation about Python packaging without coming across references to `setup.py` and `distutils`, and the terminology can get muddled. `develop` and "Development Mode" are common in the `setuptools` docs. `-e` and "editable installation" are `pip` terms, FWIW.

Reply:
@eirrgang Yes this is very true, I've been reading [Bernat Gabor's posts](https://bernat.tech/posts/) and watching some of his videos related to PEP517/518. From what I know there is a traction to move from `setup.py` to `pyproject.toml`. Few tools such as Flit and Poetry began to adopt this. And I also personally find these tools (especially Poetry) promising, but it will take some time before it is widely accepted in Python community.

As for now, IMHO I think it is still best to go with pip install as the PEP517/518 implementation is still work in progress and yet to be matured.